src/cforth/decomp2.fth

```diff
@@ -182,12 +182,11 @@ variable extent  extent off
    end-breaks @  breaks /breaks +  >=        ( adr,type full? )
    abort" Decompiler table overflow"         ( adr,type )
    end-breaks @ breaks >  if                 ( adr,type )
-      over end-breaks @ /n 2* - >r r@ 2@     ( adr,type  adr prev-adr,type )
+      over end-breaks @ /n 2* - dup >r 2@    ( adr,type  adr prev-adr,type )
       ['] .endof  =  -rot  =  and  if        ( adr,type )
-	 r@ 2@  2swap  r> 2!                 ( prev-adr,type )
-      else                                   ( adr,type )
-	 r> drop                             ( adr,type )
+         r@ 2@  2swap  r@ 2!                 ( prev-adr,type )
       then                                   ( adr,type )
+      r> drop                                ( adr,type )
```

Comment:
I hope you aren't going to continue to swamp me with little tweaks of this magnitude...